[ZBX-21373] counted proc args on openbsd manually instead of using a syscall

Index: src/libs/zbxsysinfo/openbsd/proc.c
--- src/libs/zbxsysinfo/openbsd/proc.c.orig
+++ src/libs/zbxsysinfo/openbsd/proc.c
@@ -146,7 +146,7 @@ static void	proc_data_free(proc_data_t *proc_data)
 static int	proc_argv(pid_t pid, char ***argv, size_t *argv_alloc, int *argc)
 {
 	size_t	sz;
-	int	mib[4];
+	int	mib[4], nargv = 0;
 
 	if (NULL == *argv)
 	{
@@ -171,11 +171,10 @@ retry:
 		return FAIL;
 	}
 
-	mib[3] = KERN_PROC_NARGV;
+	while (NULL != (*argv)[nargv])
+		nargv++;
 
-	sz = sizeof(int);
-	if (0 != sysctl(mib, 4, argc, &sz, NULL, 0))
-		return FAIL;
+	*argc = nargv;
 
 	return SUCCEED;
 }
